scripts/jsondeps.gradle

```diff
@@ -127,33 +127,53 @@ allprojects {
+                try {
+                    // We know that we are working against gradle version greater or equal to v3.3
+                    // So, configuration has resolvable property. If the configuration is not resolvable return null.
+                    if (config.respondsTo("isCanBeResolved")) {
+                        if (config.isCanBeResolved() == false) {
+                            println "DEBUG: Configuration is not resolvable"
+                            return null
+                        }
+                    }
+
+                    // At this point, we are working against gradle: 
+                    // - v3.3 or greater, and configuration is resolvable.
+                    // - v3.2.1 or lower, and configuration does not have canResolve property and are inherently resolvable.
+                    configsDeps = configToKeyValue config
+                    println configsDeps
+                    println "DEBUG: Configuration is resolved"
+                } catch (Exception ignored) {
+                    println "DEBUG: An exception occurred"
+                    println ignored
+                    ignored.printStackTrace()
+                }
+                return configsDeps
+            }
+
             // project: https://docs.gradle.org/current/javadoc/org/gradle/api/Project.html
             def projectToJSON = { project ->
                 def jsonConfigs = []
                 project.configurations.each { config ->
                     println "DEBUG: Trying to resolve configuration"
                     println config
-                    try {
-                        // Some configurations aren't meant to be resolved,
-                        // because they're just meant to be containers of
-                        // dependency constraints.
-                        //
-                        // See also:
-                        // - https://discuss.gradle.org/t/what-is-a-configuration-which-cant-be-directly-resolved/30721
-                        // - https://docs.gradle.org/current/userguide/declaring_dependencies.html#sec:resolvable-consumable-configs
-                        if (config.canBeResolved) {
-                            println "DEBUG: Configuration is resolvable"
-                            def result = configToKeyValue config
-                            println "DEBUG: Configuration resolved"
-                            println result
-                            jsonConfigs << result
-                        } else {
-                            println "DEBUG: Configuration is not resolvable"
-                        }
-                    } catch (Exception ignored) {
-                        println "DEBUG: An exception occurred"
-                        println ignored
-                        ignored.printStackTrace()
+                    def jsonConfigWithDeps = getResolvedConfigDepsOrEmptyOf config
+                    if (jsonConfigWithDeps != null) {
```

Comment:
[nit] Might be a tiny bit safer to return `[]` instead of `null` on empty, then we don't need to null-check the result.

Reply:
I'm leaning towards keeping `null` instead of using `flatten` and `filter`. I prefer not to do unnecessary transformations (if it can be avoided), and keeping the code for quick readability for even groovy.